Add set config

StephaneTriomphe opened the pull request.

StephaneTriomphe committed:
Add support of set_config

GwendalRaoul commented on lib/wpc_proto/internal_modules/proto_config.c:
```diff
@@ -276,7 +276,253 @@ bool Proto_config_init(void)
+    // check any config change and apply them
+    wp_SinkNewConfig * cfg = &req->config;
+
+    //cfg->sink_id[16]; not used
```
Should be checked against the only SINK_ID we have (from common).
If not the same, INVALID_SINK_ID should be returned

StephaneTriomphe replied:
ok

StephaneTriomphe replied:
done

GwendalRaoul commented on lib/wpc_proto/internal_modules/proto_config.c:
```diff
@@ -276,7 +276,253 @@ bool Proto_config_init(void)
+        app_role_t options = 0;
+        if (cfg->node_role.flags_count > 0)
+        {
+            options |= (cfg->node_role.flags[0] == wp_NodeRole_RoleFlags_LOW_LATENCY)
```
Small function for that? It is duplicated :-) And technically we could add new options later on (probably never :-) ) but roles_count may be 3.

StephaneTriomphe replied:
I started to use small fonction for get config, so I did the same here for get config (the other direction). But why duplicated, each one is called only once, no ?
If we want to support flag APP_ROLE_OPTION_RELAY, we need to add wp_NodeRole_RoleFlags_RELAY (but it's deprecated). As there is no formula to convert flags, is we want to support 3 flags, we need to change the code.
I changed the static check to remain compatible with 3 flags, and just alert if we have room for one flag.

GwendalRaoul commented on lib/wpc_proto/internal_modules/proto_config.c:
```diff
@@ -276,7 +276,253 @@ bool Proto_config_init(void)
                                                wp_SetConfigResp *resp)
 {
-    return APP_RES_PROTO_NOT_IMPLEMENTED;
+    app_res_e res;
```
To mimic the linux gateway behavior, you should automatically stop the stack and restart it automatically at the end if (and only if) one of the setting to be changed required the stack to be stopped to be changed.

StephaneTriomphe replied:
How could I know which one requires stop ?

StephaneTriomphe replied:
Done

GwendalRaoul committed:
Move StatusEvent implementation inside the config module

GwendalRaoul committed:
Use real network address in data notif

-Add an interface in proto_config module to get network address
-make Proto_config_on_stack_boot_status an internal function (because it is one)

StephaneTriomphe committed:
Adapt to new app_config_data definition

After the remove of fixed size option
